cpp/ruapu.hpp

```diff
@@ -0,0 +1,39 @@
+#include <algorithm>
```

Comment:
Please add header guards.

```suggestion
#ifndef RUAPU_HPP
#define RUAPU_HPP
#include <algorithm>
```

and corresponding `#endif` at the end of the file.